src/query.c

```diff
@@ -830,6 +835,24 @@ static IndexIterator *Query_EvalVectorNode(QueryEvalCtx *q, QueryNode *qn) {
   if (!fs || !FIELD_IS(fs, INDEXFLD_T_VECTOR)) {
     return NULL;
   }
+
+  if (qn->opts.distField) {
+    // Since the KNN syntax allows specifying the distance field in two ways (...=>[KNN ... AS <dist_field>] and
+    // ...=>[KNN ...]=>{$YIELD_DISTANCE_AS:<dist_field>), we validate that we got it only once.
+    char default_score_field[strlen(qn->vn.vq->property) + 9]; // buffer for __<field>_score
+    sprintf(default_score_field, "__%s_score", qn->vn.vq->property);
```

Comment:
I don't recall using sprintf in RediSearch. Why do we need it here?

Reply:
Its for supporting an edge case - with the new query attributes syntax, one can define the distance field twice as following:
`*=>[KNN 10 @v $b AS dist1]=>{$yield_distance_as:dist2}`. Hence, we return an error if the distance field was defined twice. However, in cases where distance field is defined only using the query attribute, we set `qn->vn.vq->scoreField` to be the default `__v_score`. In that case only, we don't return an error, but override the default score field.
`sprintf` is used to avoid allocations (save the default score field on the stack instead)

Reply:
Let's have @filipecosta90 test the performance impact.

Reply:
I see we use it all over the project.